doctr/io/pdf.py

```diff
@@ -41,8 +41,5 @@ def read_pdf(
         raise TypeError("unsupported object type for argument 'file'")
 
     # Rasterise pages to PIL images with pypdfium2 and convert to numpy ndarrays
-    pdf = pdfium.PdfDocument(file, password=password)
-    pages = [np.asarray(img) for img in pdf.render_topil(scale=scale, **kwargs)]
-    pdf.close()
-
-    return pages
+    with pdfium.PdfDocument(file, password=password) as pdf:
+        return [np.asarray(img) for img in pdf.render_topil(scale=scale, **kwargs)]
```

Comment:
I'm not certain what impact a `return` statement inside the context manager might have. Will this actually call `__close__()`? To be safe, I'd rather cleanly assign  `pages = [...]` and then put the return outside the context manager.

Reply:
Ah, according to [this thread](https://stackoverflow.com/questions/9885217/in-python-if-i-return-inside-a-with-block-will-the-file-still-close), it's safe to do this. You never stop learning!

Reply:
Should be fine close is called like in a finally block so after returning :)